src/toil/batchSystems/mesos/batchSystem.py

```diff
@@ -149,11 +149,16 @@ def __init__(self, config, maxCores, maxMemory, maxDisk):
         self.lastTimeOfferLogged = 0
         self.logPeriod = 30  # seconds
 
+        self.ignoredNodes = set()
+
         self._startDriver()
 
     def setUserScript(self, userScript):
         self.userScript = userScript
 
+    def ignoreNode(self, nodeAddress):
+        self.ignoredNodes.add(nodeAddress)
+
```

Comment:
I like this ignored node stuff. But could we add a method to *remove* ignored nodes to AbstractBatchSystem, and call it from clusterScaler at the same time it removes a node from its ignoredNodes set? (clusterScaler.py line 597 as of right now.) It's not out of the question that an instance could be marked to be ignored, terminate, then hours later a new instance might be launched that happens to get the same IP.